pkg/chaincode/lifecycle.go

```diff
@@ -85,3 +86,40 @@ func (d *Definition) validate() error {
 	}
 	return nil
 }
+
+func CreateChaincodeDefinition(channelName, packageID, name, version, endorsementPlugin, validationPlugin, signaturePolicy, channelConfigPolicy string, seq int64, initflag bool, pollections *peer.CollectionConfigPackage) (*Definition, error) {
```

Comment:
This constructor function has a large number of parameters, and many of them are actually optional. For usability, I wonder if a better approach would be to refactor this to just create the content for ValidationParameter? The Definition would then be created as normal, ignoring any unnecessary fields.

```suggestion
func NewSignaturePolicyEnvelope(policy string) (*cb.SignaturePolicyEnvelope, error) {
```

Or, if the signature policy code does not have dependencies on internals of the chaincode package, it could even be its own package, used something like:

```go
policy, err := signaturepolicy.FromString(policyString)
```

Reply:
The problem is that we have both channel level policy and signature policy, hence I will update to
```golang
policy, err := signaturepolicy.FromString(signaturepolicy, channelPolicy)
```